Assets/00_Content/Scripts/Audio/AudioManager.cs

```diff
@@ -67,32 +76,62 @@ public class AudioManager : SingletonBehaviour<AudioManager> {
 		/// <param name="fadeDuration">The duration of the fade.</param>
 		/// <param name="restart">Whether to restart playback if the same track is already playing.</param>
 		/// <param name="loop">Whether to loop the track.</param>
-		public void PlayMusic(AudioClip musicClip, FadeKind fade = FadeKind.NoFade, float fadeDuration = 0f, bool restart = false, bool loop = true) {
-			if (!restart && musicSource.clip == musicClip && musicSource.isPlaying)
-				return;
+		public void PlayMusic(AudioClip musicClip, FadeKind fade = FadeKind.NoFade, float fadeDuration = 0f,
+		                      bool restart = false, bool loop = true) {
+			PlayMusic(null, musicClip, fade, fadeDuration, restart, loop);
+		}
+
+		public void PlayMusic(MusicCue cue, FadeKind fade = FadeKind.NoFade, float fadeDuration = 0f,
+		                      bool restart = false, bool loop = true) {
+			if (cue != null)
+				PlayMusic(cue.introClip, cue.mainClip, fade, fadeDuration, restart, loop);
+			else
+				PlayMusic(null, null, fade, fadeDuration, restart, loop);
+		}
+
+		private void PlayMusic(AudioClip introClip, AudioClip musicClip, FadeKind fade = FadeKind.NoFade,
+		                       float fadeDuration = 0f, bool restart = false, bool loop = true) {
+			if (!restart
+				&& musicIntroSource.clip == introClip && musicSource.clip == musicClip
+				&& (musicIntroSource.isPlaying || musicSource.isPlaying)) return;
 
 			StopFadingMusic();
 
+			if (introClip == null && musicClip == null) {
+				StopMusic(fade == FadeKind.OutIn ? fadeDuration : 0f);
+				return;
+			}
+
 			switch (fade) {
 				case FadeKind.OutIn:
```

Comment:
This is just a small thing, but would it be possible to rename the states of the enum "FadeKind"? Because the state "OutIn" is confusing to me every time I read it. If not that, would it be possible to add a summary of tooltip to the enum that reminds the reader about what different states it can have, and what each of them mean?